lib/fastlane/actions/sync_build_number_to_git.rb

```diff
@@ -0,0 +1,38 @@
+module Fastlane
+  module Actions
+    class SyncBuildNumberToGitAction < Action
+      def self.is_git?
+        Actions.sh 'git rev-parse HEAD'
+        return true
+      rescue
+        return false
+      end
+
+      def self.run(params)
+        if is_git?
+          command = 'git rev-list HEAD --count'
+        else
+          raise "Not in a git repository."
+        end
+        build_number = (Actions.sh command).strip
+        Fastlane::Actions::IncrementBuildNumberAction.run(build_number: build_number)
```

Comment:
How about instead of actually setting the build number here, you just return the number of commits and rename the action to `number_of_commits`?

This way you could use it like this:

```ruby
build_number = number_of_commits
increment_build_number(build_number: build_number)
```

This has many advantages, including the possibility for the user to pass a custom project path to the `increment_build_number` action.